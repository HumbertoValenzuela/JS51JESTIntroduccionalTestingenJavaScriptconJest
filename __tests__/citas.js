// 8. Testing al Proyecto de Citas - Agregar Cita
import Citas from '../js/classes/Citas';

describe('Probar la clase nueva de Citas', () => {

    // Utilizar la clase de Citas
    const citas = new Citas();
    // Para utilizar las fechas y sean iguales para agregar update delete. notar que cuando se agreguen
    // más casos se deberá actualizar npm test -- -u debido a id cambiará
    const id = Date.now();
    test('Agregar una nueva cita', () => {

        const citaObj = {
            id,
            mascota: 'java',
            propietario: 'Humberto',
            telefono: '1111',
            fecha: '03-12-2021',
            hora:'12:48',
            sintomas: 'Solo Duerme' 
        };

        //  citaObj.id = id; funciona

        // metodo que vamos a probar
        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();

    });

    // 9. Testing al Proyecto de Citas - Actualizar y Eliminar
    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'java Nuevo Nombre',
            propietario: 'Humberto',
            telefono: '1111',
            fecha: '03-12-2021',
            hora:'12:48',
            sintomas: 'Solo Duerme' 
        };
        // Método que se utiliza para actualizar
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
});