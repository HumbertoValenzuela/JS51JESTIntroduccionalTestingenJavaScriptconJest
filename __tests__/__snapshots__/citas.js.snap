// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase nueva de Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "03-12-2021",
      "hora": "12:48",
      "id": 1620061871011,
      "mascota": "java Nuevo Nombre",
      "propietario": "Humberto",
      "sintomas": "Solo Duerme",
      "telefono": "1111",
    },
  ],
}
`;

exports[`Probar la clase nueva de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "03-12-2021",
      "hora": "12:48",
      "id": 1620061871011,
      "mascota": "java",
      "propietario": "Humberto",
      "sintomas": "Solo Duerme",
      "telefono": "1111",
    },
  ],
}
`;

exports[`Probar la clase nueva de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;
